SonarQube.Client/Api/V9_4/GetSonarLintEventStream.cs

```diff
@@ -42,6 +46,42 @@ internal class GetSonarLintEventStream : RequestBase<Stream>, IGetSonarLintEvent
                 MediaTypeWithQualityHeaderValue.Parse("text/event-stream")
             };
 
+        public override async Task<Stream> InvokeAsync(HttpClient httpClient, CancellationToken token)
+        {
+            var requestAttempts = 0;
+
+            return await InvokeWithRetriesAsync();
+
+            // Code on the java side: https://github.com/SonarSource/sonarlint-core/blob/4f34c7c844b12e331a61c63ad7105acac41d2efd/server-api/src/main/java/org/sonarsource/sonarlint/core/serverapi/stream/EventStream.java#L101
+            async Task<Stream> InvokeWithRetriesAsync()
+            {
+                do
+                {
+                    requestAttempts++;
+
+                    var result = await InvokeUncheckedAsync(httpClient, token);
+
+                    if (ShouldRetry(result) && requestAttempts < MaxNumberOfRequestAttempts)
+                    {
+                        continue;
+                    }
+
+                    result.EnsureSuccess();
+
+                    return result.Value;
+                } while (true);
```

Comment:
A suggestion, but not a strong one: maybe we can move the exit condition to the while statement argument instead of `true`, it can be easier to read in some cases.